nebula_ros/src/continental/continental_ars548_decoder_wrapper.cpp

```diff
@@ -151,12 +151,14 @@ void ContinentalARS548DecoderWrapper::on_config_change(
 {
   std::lock_guard lock(mtx_driver_ptr_);
   initialize_driver(new_config_ptr);
+  std::unique_lock lock_cfg(mtx_config_ptr_);
```

Comment:
Locking on the driver and config mutex separately is deadlock-prone, as a future maintainer might lock them elsewhere in a different order. I would suggest using a `std::scoped_lock` which takes multiple mutexes and performs deadlock avoidance.